### DISCUSSION TITLE
SignIn: Different callbackUrl for returning users vs. new users

### DISCUSSION THREAD

**Discussion by goerlitz** [nextauthjs/next-auth#9066] [Help]
I'm working on a SaaS application where new users should be redirected to an onboarding page (aka Getting Started page) after successful SignIn with social provider or using a magic link.

I'm used to having separate login and registration pages but it seems that the default SingIn page can cover both cases as I can check for a new user in the SingIn callback like this:

```ts
callbacks: {
  async signIn({ user, account, profile, email, credentials }) {

    // using Prisma adapter (https://authjs.dev/reference/adapter/prisma)
    const knownUser = await prisma.user.findFirst({
      where: {
        email: user.email,
      },
    });

    if (!knownUser) {
      // change callbackUrl to redirect to new user onboarding
    }

    return true;
  },
},
```

However, I can't figure out how to modify the callback URL.

**Question:** What is the best practice to handle these two cases?

Of course, I could implement two custom pages for SingIn and SignUp (and specify a different callbackUrl in the latter case). But maybe there is an easier solution.

**Comment by sdgoldstein** [comment]
@goerlitz I'm trying to achieve the same behavior and I also don't see a way to do it.  The only thing I can think of is having it callback to a single URL which checks the user state and then redirects if the user should be onboarded.

Did you come up with any other options?
